### PR TITLE
feat: auth return url

### PR DESCRIPTION
**Pull request by alexanderallen-aa**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

### CODE REVIEW

File: packages/api/src/api/auth/get.ts [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-  client_id: env.slackClientID,
-});
-const fullLink = `${slackAuthBaseUrl}${queryArgs.toString()}`;
 
 export const get = async (req: Request, res: Response) => {
-  res.redirect(fullLink);
+  const { [Config.global.authReturnUriOverride]:returnTo } = req.query as Record<string, string>;
+
+  const queryArgs = new URLSearchParams({
+    redirect_uri: `${slackCallbackUrl}?${returnTo}`,
```
Comment (SpencerKaiser) [inline packages/api/src/api/auth/get.ts:13]:
```suggestion
    redirect_uri: `${slackCallbackUrl}?${Config.global.authReturnUriOverride}=${returnTo}`,
```

File: packages/shared/src/config/global.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
 export const global = {
   appName: 'Hangar',
+  authReturnUriOverride: 'returnTo',
```
Comment (SpencerKaiser) [inline packages/shared/src/config/global.ts:3]: I'd prefer something like `authReturnToQueryKey`... thoughts?

Comment (alexanderallen-aa) [reply packages/shared/src/config/global.ts:3]: i mean, i would not understand that... `returnTo` is ambiguous, and i wouldn't even know what to make of `QueryKey`; whereas `returnUri` calls out the purpose, and `override` signals that it is optional.

File: packages/api/src/api/auth/get.ts (hunk 1 above)
Comment (alexanderallen-aa) [reply packages/api/src/api/auth/get.ts:13]: thanks for catching that

File: packages/api/tests/api/auth/get.test.ts [hunk 3]
```diff
@@ -1,14 +1,26 @@
...
 import { createMockResponse } from '../../testUtils/expressHelpers/createMockResponse';
 
+const slackAuthBaseUrl: string =
+  'https://slack.com/openid/connect/authorize?scope=openid%20email%20profile&response_type=code&';
+
+const returnTo = '/api/expoJudgingSession';
+const redirect_uri = encodeURIComponent(
+  `/api/auth/callback/slack?${Config.global.authReturnUriParamName}=${returnTo}`,
+);
+// const redirect_uri = `${slackCallbackUrl}=${Config.global.authReturnUriParamName}=${encodeURIComponent( '/api/auth/callback/slack')}`
```
Comment (SpencerKaiser) [inline packages/api/tests/api/auth/get.test.ts:13]:
```suggestion
```

File: packages/api/tests/utils/authenticateUser.test.ts [hunk 4]
```diff
@@ -72,4 +72,20 @@ describe('authenticate user', () => {
...
+      firstName: 'a',
+      lastName: 'b',
+      returnTo: '/test',
+    };
+
+    await authenticateUser({ data, req, res } as any);
+
+    expect(res.redirect).toHaveBeenCalledTimes(1);
+    expect(res.redirect).toHaveBeenCalledWith('/test');
+  });
```
Comment (SpencerKaiser) [inline packages/api/tests/utils/authenticateUser.test.ts:90]: We're missing one test to check if the path provided does not start with `/`

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]

**Review by SpencerKaiser — APPROVED** [review]

**Pull request merged**